src/System.Private.CoreLib/src/System/Delegate.cs

```diff
@@ -372,19 +373,19 @@ public static Delegate CreateDelegate(Type type, object target, string method, b
         }
 
         // V1 API.
-        public static Delegate CreateDelegate(Type type, Type target, string method)
+        public static Delegate? CreateDelegate(Type type, Type target, string method)
         {
             return CreateDelegate(type, target, method, false, true);
         }
 
         // V1 API.
-        public static Delegate CreateDelegate(Type type, Type target, string method, bool ignoreCase)
+        public static Delegate? CreateDelegate(Type type, Type target, string method, bool ignoreCase)
         {
             return CreateDelegate(type, target, method, ignoreCase, true);
         }
 
         // V1 API.
-        public static Delegate CreateDelegate(Type type, Type target, string method, bool ignoreCase, bool throwOnBindFailure)
+        public static Delegate? CreateDelegate(Type type, Type target, string method, bool ignoreCase, bool throwOnBindFailure)
```

Comment:
This can return null if the binding fails and throwOnBindingFailure is false.  But the above methods I commented on all pass true.

Reply:
Right, but the compiler can't trace that deeply. Should it? How far do we expect it to infer?

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDk4BstWFiAEQHsBzMY4gEwBoqQBqAHwGIA7bXfCAAgjbwEAsACgAAgAYeYgIwBuUWIDM0gEw8AwjwDeonvukqA8sABWEAMYweAWQAUASh16DrsQHYe6O2wgB3HmIzSxhHGh4YAAsEYj8jNkwcEBgEOAgHBRFXAF8XfTyeAAcEAEsAN1heE3MrAH4vO2qQwPDgSiwI6Nj4xKwnXSzXAxKwHjtiHgBeSZ4OHB4AMgXOmLiEzgcCoYMo1dn/HgBRAA8LCEKYEuI2R0zt/MHtj0C7g1yRbKA===

Reply:
> Right, but the compiler can't trace that deeply. Should it? How far do we expect it to infer?

No, but you can.  Our public annotations should not depend on what the compiler can infer.

Reply:
You mean add `!` to the calls that pass true? OK, I did that. It is better.